geopandas/testing.py

```diff
@@ -123,15 +123,28 @@ def assert_geoseries_equal(
     if not check_crs:
         with warnings.catch_warnings():
             warnings.filterwarnings("ignore", "CRS mismatch", UserWarning)
-            if check_less_precise:
-                assert geom_almost_equals(left, right)
-            else:
-                assert geom_equals(left, right)
+            check_equality(left, right, check_less_precise)
+    else:
+        check_equality(left, right, check_less_precise)
+
+
+def check_equality(left, right, check_less_precise):
+    if check_less_precise:
+        if not geom_almost_equals(left, right):
+            unequal_geoms = left[left.geom_almost_equals(right)]
```

Comment:
```suggestion
            unequal_geoms = left[~left.geom_almost_equals(right)]
```
We need to get unequal here, so the mask should be inverted.

Reply:
Indeed, sorry i overlooked there